File: problems/2465-number-of-distinct-averages/solution.py
class Solution:
    def distinctAverages(self, nums: list[int]) -> int:
        nums.sort()
        averages = set()
        left, right = 0, len(nums) - 1
        while left < right:
            averages.add((nums[left] + nums[right]) / 2)
            left += 1
            right -= 1
        return len(averages)

File: problems/2465-number-of-distinct-averages/solution.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int distinctAverages(int[] nums) {
        Arrays.sort(nums);
        Set<Double> averages = new HashSet<>();
        int left = 0;
        int right = nums.length - 1;
        while (left < right) {
            double average = (double) (nums[left] + nums[right]) / 2;
            averages.add(average);
            left++;
            right--;
        }
        return averages.size();
    }
}